velox/dwio/parquet/tests/ParquetTpchTestBase.h

```diff
@@ -74,28 +81,36 @@ class ParquetTpchTestBase : public testing::Test {
     auto task = assertQuery(tpchPlan, duckDbSql, sortingKeys);
   }
 
-  void TearDown() {
+  static void TearDownTestSuite() {
     connector::unregisterConnector(kHiveConnectorId);
-    unregisterParquetReaderFactory();
+    velox::parquet::unregisterParquetReaderFactory();
   }
 
  private:
   /// Write TPC-H tables as a Parquet file to temp directory in hive-style
   /// partition
-  void saveTpchTablesAsParquet() {
-    constexpr int kRowGroupSize = 10'000;
-    const auto tableNames = tpchBuilder_.getTableNames();
-    for (const auto& tableName : tableNames) {
+  static void saveTpchTablesAsParquet(memory::MemoryPool* pool) {
+    for (const auto& table : tpch::tables) {
+      auto tableName = toTableName(table);
       auto tableDirectory =
           fmt::format("{}/{}", tempDirectory_->path, tableName);
-      fs::create_directory(tableDirectory);
-      auto filePath = fmt::format("{}/file.parquet", tableDirectory);
-      auto query = fmt::format(
-          fmt::runtime(kDuckDbParquetWriteSQL_.at(tableName)),
-          tableName,
-          filePath,
-          kRowGroupSize);
-      duckDb_->execute(query);
+      auto tableSchema = tpch::getTableSchema(table);
+      auto names = tableSchema->names();
+      auto plan =
+          PlanBuilder().tableScan(table, std::move(names), 0.01).planNode();
+      auto split =
+          exec::Split(std::make_shared<connector::tpch::TpchConnectorSplit>(
+              kTpchConnectorId, 1, 0));
+
+      auto rows = AssertQueryBuilder(plan).splits({split}).copyResults(pool);
+      duckDb_->createTable(tableName, {rows});
```

Comment:
Why do we need this? Can we not create single plan like this:

```
  PlanBuilder()
    .tableScan(table, std::move(names), 0.01)
    .tableWrite(tableDirectory, dwio::common::FileFormat::PARQUET)
    .planNode()
```

Reply:
We need to use the Velox tpch connector to generate data to create DuckDB tables used by
the `DuckDbQueryRunner`.

The same generated data is also used by the `TpchQueryBuilder`.